kobo/pkgset.py

```diff
@@ -229,6 +229,9 @@ def __getitem__(self, name):
     def __setitem__(self, name, value):
         self.file_cache[os.path.abspath(name)] = value
 
+    def __contains__(self, item):
```

Comment:
I must say, the mismatch between `getitem/setitem` and `contains` is concerning.

Keys are converted into absolute paths in the getter/setter but they are NOT converted during a `contains` check.
This means the basic contract of "inserting an object into a container means the container contains it" is violated:

```python
from kobo.pkgset import FileCache

fc = FileCache()

fc['/y'] = object()
# Is the thing we just inserted in there?
# Of course it is...
assert '/y' in fc

# How about now?
fc['x'] = object()
# Nope - CRASH! I asked for 'x' but it stored '$PWD/x'
# and did not do the same transformation during 'contains'.
assert 'x' in fc  
```

I initially thought that your change introduced this problem, however the problem is already present on the current `foo in self.file_cache` path making use of `__iter__`, the code just wasn't visible.

I guess we should not attempt to fix that in this change, since it might be backwards-incompatible.
But if you are looking at code using this class, maybe you want to take a look at it with this bug in mind.
Any code using this class and not providing absolute paths as input has the potential to behave very strangely.